drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,599 @@
+			    sl_conn.bssid[1], sl_conn.bssid[2],
+			    sl_conn.bssid[3], sl_conn.bssid[4],
+			    sl_conn.bssid[5]);
+
+		/* Continue the notification callback chain... */
+		sl_conn.error = 0;
+		nwp.cb(SL_WLAN_EVENT_CONNECT, &sl_conn);
+		break;
+
+	case SL_WLAN_EVENT_DISCONNECT:
+		CLR_STATUS_BIT(nwp.status, STATUS_BIT_CONNECTION);
+		CLR_STATUS_BIT(nwp.status, STATUS_BIT_IP_ACQUIRED);
+		CLR_STATUS_BIT(nwp.status, STATUS_BIT_IPV6_ACQUIRED);
+
+		event_data = &wlan_event->Data.Disconnect;
+
+		/* If the user has initiated 'Disconnect' request,
+		 * 'reason_code' is SL_WLAN_DISCONNECT_USER_INITIATED
+		 */
+		if (SL_WLAN_DISCONNECT_USER_INITIATED ==
+		    event_data->ReasonCode) {
+			SYS_LOG_INF("\n[WLAN EVENT] "
+				    "Device disconnected from the AP: %s,\n\r"
+				    "BSSID: %x:%x:%x:%x:%x:%x on application's"
+				    " request",
+				    event_data->SsidName, event_data->Bssid[0],
+				    event_data->Bssid[1], event_data->Bssid[2],
+				    event_data->Bssid[3], event_data->Bssid[4],
+				    event_data->Bssid[5]);
+			sl_conn.error = 0;
+		} else {
+			SYS_LOG_ERR("\n[WLAN ERROR] "
+				    "Device disconnected from the AP: %s,\n\r"
+				    "BSSID: %x:%x:%x:%x:%x:%x on error: %d",
+				    event_data->SsidName, event_data->Bssid[0],
+				    event_data->Bssid[1], event_data->Bssid[2],
+				    event_data->Bssid[3], event_data->Bssid[4],
+				    event_data->Bssid[5],
+				    event_data->ReasonCode);
+			sl_conn.error = event_data->ReasonCode;
+		}
+
+		memset(&(sl_conn.ssid), 0x0, sizeof(sl_conn.ssid));
+		memset(&(sl_conn.bssid), 0x0, sizeof(sl_conn.bssid));
+
+		/* Continue the notification callback chain... */
+		nwp.cb(SL_WLAN_EVENT_DISCONNECT, &sl_conn);
+		break;
+
+	case SL_WLAN_EVENT_STA_ADDED:
+		memcpy(&(sl_conn.bssid), wlan_event->Data.STAAdded.Mac,
+		       SL_WLAN_BSSID_LENGTH);
+		SYS_LOG_INF("\n[WLAN EVENT] STA was added to AP: "
+			    "BSSID: %x:%x:%x:%x:%x:%x",
+			    sl_conn.bssid[0], sl_conn.bssid[1],
+			    sl_conn.bssid[2], sl_conn.bssid[3],
+			    sl_conn.bssid[4], sl_conn.bssid[5]);
+		break;
+	case SL_WLAN_EVENT_STA_REMOVED:
+		memcpy(&(sl_conn.bssid), wlan_event->Data.STAAdded.Mac,
+		       SL_WLAN_BSSID_LENGTH);
+		SYS_LOG_INF("\n[WLAN EVENT] STA was removed from AP: "
+			    "BSSID: %x:%x:%x:%x:%x:%x",
+			    sl_conn.bssid[0], sl_conn.bssid[1],
+			    sl_conn.bssid[2], sl_conn.bssid[3],
+			    sl_conn.bssid[4], sl_conn.bssid[5]);
+
+		memset(&(sl_conn.bssid), 0x0, sizeof(sl_conn.bssid));
+		break;
+	default:
+		SYS_LOG_ERR("\n[WLAN EVENT] Unexpected event [0x%lx]",
+			    wlan_event->Id);
+		break;
+	}
+}
+
+/**
+ *  @brief SimpleLinkNetAppEventHandler
+ *
+ *  This handler gets called whenever a Netapp event is reported
+ *  by the host driver / NWP.
+ *
+ *  @note See the CC3120/CC3220 NWP programmer's guide (SWRU455)
+ *	  section 5.7.
+ */
+void SimpleLinkNetAppEventHandler(SlNetAppEvent_t *netapp_event)
+{
+	u32_t i = 0;
+
+	SlIpV4AcquiredAsync_t *event_data = NULL;
```

Comment:
revert the order of these 2 declarations. And don't pre-assign 'i': your for loop is doing it anyway

Reply:
Thanks.